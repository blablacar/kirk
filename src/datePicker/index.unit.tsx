import React from 'react'
import { mount } from 'enzyme'
import moment from 'moment'

import DatePicker, { HORIZONTAL_ORIENTATION } from 'datePicker'

describe('DatePicker', () => {
  it('Should retrieve the react dates contants from the kirk component', () => {
    expect(HORIZONTAL_ORIENTATION).toBe('horizontal')
  })

  it('Should render the DatePicker with an initialDate', () => {
    const initialDate = moment('2020-01-01')
    const wrapper = mount(<DatePicker name="datepicker" initialDate={initialDate} />)
    const dayPickerSingleDateController = wrapper.find('DayPickerSingleDateController')
    expect(dayPickerSingleDateController.prop('date')).toBe(initialDate)
  })

  describe('#onChange', () => {
    it('Should return the date in the format `YYYY-MM-DD`', () => {
      const onChange = jest.fn()
      const date = '2020-01-01 04:42:42'
      const wrapper = mount(<DatePicker name="datepicker" onChange={onChange} />)
      wrapper.instance().onDateChange(moment(date))
      expect(onChange).toHaveBeenCalledWith({ name: 'datepicker', value: '2020-01-01' })
    })
  })
})
