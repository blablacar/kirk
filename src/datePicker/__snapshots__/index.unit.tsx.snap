// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renderCaption Should render the given month title 1`] = `
<div
  className="kirk-datepicker-month-caption DayPicker-Caption"
>
  January 
</div>
`;

exports[`DatePicker renderCaption Should render the given month title with year if it is not the current year 1`] = `
<div
  className="kirk-datepicker-month-caption DayPicker-Caption"
>
  January 2050
</div>
`;

exports[`DatePicker renderNavbar Should render the previous/next buttons in horizontal mode 1`] = `
<div
  className=""
>
  <button
    className="kirk-button kirk-button-tertiary kirk-button-bubble kirk-datepicker-previous-month"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="kirk-icon"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <g
        fill="none"
        stroke="#00AFF5"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeMiterlimit="10"
        strokeWidth="2"
      >
        <path
          d="M22 12H2"
        />
        <path
          d="M9 19l-7-7 7-7"
        />
      </g>
    </svg>
  </button>
  <button
    className="kirk-button kirk-button-tertiary kirk-button-bubble kirk-datepicker-next-month"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="kirk-icon"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <g
        fill="none"
        stroke="#00AFF5"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeMiterlimit="10"
        strokeWidth="2"
        transform="rotate(180 12 12)"
      >
        <path
          d="M22 12H2"
        />
        <path
          d="M9 19l-7-7 7-7"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`DatePicker renderNavbar Should render the weekdays in vertical mode 1`] = `
<div
  aria-hidden="true"
  className="kirk-datepicker-sticky-weekdays"
  style={
    Object {
      "top": "0px",
    }
  }
>
  <div
    className="kirk-datepicker-sticky-weekdaysrow"
  >
    <div
      className="kirk-datepicker-sticky-weekday"
    >
      Su
    </div>
    <div
      className="kirk-datepicker-sticky-weekday"
    >
      Mo
    </div>
    <div
      className="kirk-datepicker-sticky-weekday"
    >
      Tu
    </div>
    <div
      className="kirk-datepicker-sticky-weekday"
    >
      We
    </div>
    <div
      className="kirk-datepicker-sticky-weekday"
    >
      Th
    </div>
    <div
      className="kirk-datepicker-sticky-weekday"
    >
      Fr
    </div>
    <div
      className="kirk-datepicker-sticky-weekday"
    >
      Sa
    </div>
  </div>
</div>
`;
